pandas/core/resample.py

```diff
@@ -1027,9 +1027,22 @@ def quantile(self, q=0.5, **kwargs):
 # downsample methods
 for method in ["sum", "prod", "min", "max", "first", "last"]:
 
-    def f(self, _method=method, min_count=0, *args, **kwargs):
+    def f(
+        self,
+        _method: str = method,
+        numeric_only: bool | lib.NoDefault = lib.no_default,
+        min_count: int = 0,
+        *args,
+        **kwargs,
+    ):
+        if numeric_only is lib.no_default:
+            if (self.obj.ndim == 1) or (_method != "sum"):
+                # For SeriesGroupBy, set the default to be False.
+                # For DataFrameGroupBy, set it to be False for methods other than `sum`.
+                numeric_only = False
```

Comment:
It looks like the numeric_only just gets passed through; is there a reason we need to overwrite with False here?

For Series, the user setting to True should raise; what is the error message when this occurs? If it is confusing, it would be better to intercept here and emit a sensible message.

Reply:
Thanks @rhshadrach. The reason for overwriting `lib.no_default` with `False` is simply to pass the current test cases. 

Since the  `numeric_only`  default value for some of the methods in the `GroupBy` class is `False` (such as `min`, `first`, `last`), if we just pass the `lib.no_default`, some of the test cases will be broken. Please see the following error message for reference.

> FAILED pandas/tests/resample/test_datetime_index.py::test_resample_datetime_values - KeyError: 'timestamp'
> FAILED pandas/tests/resample/test_resampler_grouper.py::test_resample_groupby_agg_object_dtype_all_nan[True] - AssertionError: DataFrame are different
> FAILED pandas/tests/resample/test_resampler_grouper.py::test_resample_groupby_agg_object_dtype_all_nan[False] - AssertionError: DataFrame are different
> FAILED pandas/tests/resample/test_timedelta.py::test_resample_timedelta_values - KeyError: 'time'


For Series, I overwrite with `False` for the same reason. Both `True` and `lib.no_default` will raise a NotImplementedError,
so the following test case will be broken.
> FAILED pandas/tests/resample/test_resample_api.py::test_resample_agg_readonly - NotImplementedError: SeriesGroupBy.last does not implement numeric_only.

Reply:
Ah, I see - groupby methods like min only take True/False and don't have lib.no_default as their default. Makes sense.